src/apic-extension/azext_apic_extension/command_patches.py

```diff
@@ -379,3 +381,65 @@ def pre_operations(self):
 
         args.apim_resource_id = (f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/"
                                  f"Microsoft.ApiManagement/service/{args.apim_name}/")
+
+        # Set api_source_type
+        args.api_source_type = "AzureApiManagement"
+
+
+@register_command(
+    "apic integration create amazon-api-gateway",
+    is_preview=True,
+)
+class CreateAmazonApiGatewayIntegration(DefaultWorkspaceParameter, CreateIntegration):
+    """Add Amazon API Gateway as API source
+
+    :example: Add Amazon API Gateway as an API source
+        az apic integration create amazon-api-gateway -g contoso-resources -n contoso
+            --integration-id sync-from-my-amazon-api-gateway
+            --access-key https://mykey.vault.azure.net/secrets/AccessKey
+            --secret-access-key https://mykey.vault.azure.net/secrets/SecretAccessKey
+            --region-name us-east-2
+    """
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        # pylint: disable=protected-access
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        # Remove the amazon_api_gateway_source parameter
+        args_schema.amazon_api_gateway_source._registered = False
+        # Remove the apim_resource parameter
+        args_schema.apim_resource_id._registered = False
+        args_schema.msi_resource_id._registered = False
+
+        # Add separate parameters for access-key, secret-access-key, and region-name
+        args_schema.access_key = AAZStrArg(
```

Comment:
Since `_create.py` defines these parameters, do we need to define these parameters here again?

Reply:
Yes. In _create.py, the definition won't show as -- format. So I re-define them here.